hiera/settings.yaml

```diff
@@ -1,75 +1,78 @@
-    password: 'password'
-    root_password: 'password'
-    log_path: '/log/database'
-
-## application logging: the possible logging levels are constrained by the
-#                      python 'logging' module.
-#
-#  @LOG_LEVEL, defines the application baseline (i.e. handler) level for logs.
-#      This means the logger level cannot exceed this baseline.
-#
-#      The following are supported 'LOG_LEVEL' values:
-#
-#      - ERROR
-#      - WARNING
-#      - INFO
-#      - DEBUG
-#
-#  Note: the specific log levels can be reviewed:
-#
-#        https://docs.python.org/2/library/logging.html#logging-levels
-#
-#  Note: when referencing a hash path value, remember to prefix the call with
-#       the above 'general: root' definition.
-#
-application:
-    model_type:
-        - svm
-        - svr
-    log_level: 'DEBUG'
-    error_log_path: '/log/application/error'
-    warning_log_path: '/log/application/warning'
-    info_log_path: '/log/application/info'
-    debug_log_path: '/log/application/debug'
+## @settings
+#
+#  This file contains configuration settings for the following:
+#
+#    - redis
+#    - database
+#    - application logging
+#
+
+## variables
+general:
+    environment: 'vagrant'
+    vagrant_implement: true
+    root: '/vagrant'
+    host: 'localhost'
+    user: 'vagrant'
+    group: 'vagrant'
+
+## redis
+redis:
+    bind_address: '127.0.0.1'
+    host: 'localhost'
+    port: 6379
+    path: '/etc/redis/redis.conf'
+
+## webserver
+webserver:
+    flask_log_path: '/log/webserver/flask.log'
+
+## application database
+#
+#  @host, defined from executed 'docker' command.
+#
+#  Note: when referencing a hash path value, remember to prefix the call with
+#       the above 'general: root' definition.
+#
+database:
+    host: 'localhost'
+    name: 'db_machine_learning'
+    provisioner: 'provisioner'
+    provisioner_password: 'password'
+    username: 'authenticated'
+    password: 'password'
+    root_password: 'password'
+    log_path: '/log/database'
+
+## application logging: the possible logging levels are constrained by the
+#                      python 'logging' module.
+#
+#  @LOG_LEVEL, defines the application baseline (i.e. handler) level for logs.
+#      This means the logger level cannot exceed this baseline.
+#
+#      The following are supported 'LOG_LEVEL' values:
+#
+#      - ERROR
+#      - WARNING
+#      - INFO
+#      - DEBUG
+#
+#  Note: the specific log levels can be reviewed:
+#
+#        https://docs.python.org/2/library/logging.html#logging-levels
+#
+#  Note: when referencing a hash path value, remember to prefix the call with
+#       the above 'general: root' definition.
+#
+application:
+    model_type:
+        - svm
+        - svr
+    log_level: 'DEBUG'
+    error_log_path: '/log/application/error'
+    warning_log_path: '/log/application/warning'
+    info_log_path: '/log/application/info'
+    debug_log_path: '/log/application/debug'
+
+crypto:
+    salt_length: '32'

```

Comment:
I'd remove the quotes around `'32'`, that way the application doesn't need to cast this to an integer.